authorization-service/authorization_service/lambda_func/basic_authorizer.py

```diff
@@ -0,0 +1,62 @@
+import os
+import base64
+
+
+def lambda_handler(event, context):
+    print(f"event: {event}")
+
+    authorization_header = event.get('authorizationToken', '')
+    if not authorization_header or ' ' not in authorization_header:
+        raise Exception('Unauthorized')
+
+    try:
+        auth_scheme, auth_b64 = authorization_header.split(' ', 1)
+
+        print(f"auth_scheme: {auth_scheme} auth_b64 {auth_b64}")
+
+        if not auth_b64 or auth_b64 == 'null' or auth_scheme != 'Basic':
+            return generate_policy('user', 'Deny', event['methodArn'])
+
+        try:
+            auth_decoded = base64.b64decode(auth_b64).decode('utf-8')
+        except Exception:
+            return generate_policy('user', 'Deny', event['methodArn'])
+
+        username, password = auth_decoded.split('=', 1)
+        expected_password = os.environ.get(username)
+        if not expected_password or expected_password != password:
+            return generate_policy(username, 'Deny', event['methodArn'])
+
+        return generate_policy(username, 'Allow', event['methodArn'])
+
+    except Exception as e:
```

Comment:
If anything except of policy is returned by lambda handler method response will return with 500 http status, no need to return http response here, returning null of throwing exception would behave in exactly the same way.

Reply:
Agree!